src/crimson/osd/osd_operations/snaptrim_event.cc

```diff
@@ -524,9 +530,17 @@ SnapTrimObjSubEvent::with_pg(
           });
         });
       });
-    }).handle_error_interruptible(PG::load_obc_ertr::all_same_way([] {
-      return seastar::now();
-    }));
+    }).handle_error_interruptible(
+      // See how classic handles this case at the beginning of
+      // PrimaryLogPG::trim_object,
+      crimson::ct_error::enoent::handle([this](auto &) {
+        logger().error("{}: saw ENOENT while loading obc,"
+                       "setting pg state to SNAPTRIM_ERROR", *this);
+        pg->get_peering_state().state_set(PG_STATE_SNAPTRIM_ERROR);
+        return seastar::now();
```

Comment:
That's what the previous comment is about. Conceptually returning `now()` informs that an error has been _handled_ but here we just repack / redirect it to a new communication channel. Communication channe that strongly glues parts of loops of `SnapTrimObjSubEvent` and `SnapTrimEvent` together.